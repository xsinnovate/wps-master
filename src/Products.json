{
  "prod1": {
    "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
    "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
    "price": 102.96,
    "pickup": 3.85
  }
}
