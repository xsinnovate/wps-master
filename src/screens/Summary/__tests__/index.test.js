import React from 'react';
import renderer from 'react-test-renderer';
import Products from '../../../Products.json';
import { Summary } from '..';

describe('<Summary />', () => {
  it('renders correctly', () => {
    const tree = renderer.create((
      <Summary
        products={[{
          ...Products.prod1,
          id: 'prod1',
          quantity: 1,
        }]}
        subtotal={100}
        savings={10}
        promo={0}
        taxes={10}
        total={100}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders promo discount line if provided', () => {
    const tree = renderer.create((
      <Summary
        products={[{
          ...Products.prod1,
          id: 'prod1',
          quantity: 1,
        }]}
        subtotal={100}
        savings={10}
        promo={10}
        taxes={10}
        total={90}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
