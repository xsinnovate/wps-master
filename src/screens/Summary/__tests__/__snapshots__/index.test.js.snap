// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 40,
        "paddingHorizontal": 20,
        "paddingTop": 75,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            Subtotal
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            $100
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  Object {
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              Pickup Savings
            </Text>
          </View>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "paddingHorizontal": 20,
                    "paddingVertical": 25,
                  }
                }
              >
                <Text>
                  Picking up your order in the store helps cut costs, and we pass the savings on to you.
                </Text>
              </View>
            </View>
          </Modal>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "600",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            -$10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            Est. taxes & fees
            

            (Based on 94085)
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            $10
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 8,
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Est. total
          </Text>
          <Text
            style={
              Object {
                "fontSize": 28,
                "fontWeight": "600",
              }
            }
          >
            $100
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text>
                See item details
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 32,
                    "fontWeight": "100",
                    "lineHeight": 32,
                    "marginLeft": 10,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "paddingVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#dfdfdf",
                "height": 1,
              }
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text>
                Apply promo code
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 32,
                    "fontWeight": "100",
                    "lineHeight": 32,
                    "marginLeft": 10,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<Summary /> renders promo discount line if provided 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingBottom": 40,
        "paddingHorizontal": 20,
        "paddingTop": 75,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            Subtotal
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            $100
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  Object {
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              Pickup Savings
            </Text>
          </View>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              accessible={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "paddingHorizontal": 20,
                    "paddingVertical": 25,
                  }
                }
              >
                <Text>
                  Picking up your order in the store helps cut costs, and we pass the savings on to you.
                </Text>
              </View>
            </View>
          </Modal>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "600",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            -$10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            Promo
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "600",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            -$10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "lineHeight": 22,
              }
            }
          >
            Est. taxes & fees
            

            (Based on 94085)
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            $10
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#dfdfdf",
              "height": 1,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 8,
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Est. total
          </Text>
          <Text
            style={
              Object {
                "fontSize": 28,
                "fontWeight": "600",
              }
            }
          >
            $90
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text>
                See item details
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 32,
                    "fontWeight": "100",
                    "lineHeight": 32,
                    "marginLeft": 10,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "paddingVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#dfdfdf",
                "height": 1,
              }
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text>
                Apply promo code
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 32,
                    "fontWeight": "100",
                    "lineHeight": 32,
                    "marginLeft": 10,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
